core/src/stream.rs

```diff
@@ -13,14 +13,59 @@ const TIMEOUT: Duration = Duration::from_millis(10_000);
 #[cfg(not(test))]
 const TIMEOUT: Duration = Duration::from_millis(1000);
 
-/// Represents messsage used by the stream
+/// Error type with generic for actual Stream/Sink error type
+#[derive(thiserror::Error, Debug)]
+pub enum Error<InternalStreamError>
+where
+    InternalStreamError: std::error::Error + Send + Sync + 'static,
+{
+    /// Error, that occurs when `recv()` timeout exceeds
```

Comment:
```suggestion
    /// `recv()` timeout exceeded
```
It's a lot shorter, and most of our variant docs must fit on a single line for the doc generator to work.